test/test_hpyerr.py

```diff
@@ -461,14 +461,20 @@ def test_HPyErr_WarnEx(self):
         """)
 
         # NOTE: trampoline is defined in support.py
+        import sys
+        ISPYPY = "__pypy__" in sys.modules
         def check_warning(arg, category, message, file):
             with warnings.catch_warnings(record=True) as warnings_list:
-                trampoline(mod.f, arg)
+                if ISPYPY:
+                    mod.f(arg)
+                else:
+                    trampoline(mod.f, arg)
                 assert len(warnings_list) == 1, str(category)
                 w = warnings_list[-1]
                 assert issubclass(w.category, category), str(category)
                 assert str(w.message) == message, str(category)
-                assert w.filename.endswith(file), str(category)
+                if not ISPYPY:
+                    assert w.filename.endswith(file), str(category)
```

Comment:
Why are the filenames different on PyPy?

Reply:
This is a quirk in the untranslated tests. Removing the change.